### PR TITLE
Adjust tsh language in regards to Webauthn

### PR DESCRIPTION
**Pull request by codingllama**
This is a collection of a few small changes related to user presentation of
WebAuthn/MFA in tsh. The intent is to make tsh language match ongoing Web UI
changes.

Changes included:

* Use the PreferredLocalMFA server-side hint to decide lingo in `tsh mfa add`
* Adjust language when prompting for challenges to mention WEBAUTHN (and replace
  U2F with the more generic MFA)
* Sort devices by name on `tsh mfa ls`
* Address origin validation TODOS

### CODE REVIEW

File: tool/tsh/mfa.go [hunk 1]
```diff
@@ -157,11 +162,33 @@ func newMFAAddCommand(parent *kingpin.CmdClause) *mfaAddCommand {
...
+			return trace.Wrap(err)
+		}
+		log.Debugf("Got server-side preferred local MFA: %v", pingResp.Auth.PreferredLocalMFA)
+		switch preferredMFA := pingResp.Auth.PreferredLocalMFA; preferredMFA {
+		case constants.SecondFactorOTP: // OTP only
+			deviceTypes = []string{totpDeviceType}
+		case constants.SecondFactorU2F: // OTP+U2F
+			deviceTypes = []string{totpDeviceType, u2fDeviceType}
+		case constants.SecondFactorWebauthn: // OTP+WebAuthn
+			deviceTypes = []string{totpDeviceType, webauthnDeviceType}
```
Comment (kimlisa) [inline tool/tsh/mfa.go:187]: hm.. i am not sure i understand why totp device is included in both?

Comment (kimlisa) [reply tool/tsh/mfa.go:187]: oh i see, if sf is `optional` or `on` it returns webauthn or u2f, but how would you know if it's only u2f or only webauthn?

Comment (codingllama) [reply tool/tsh/mfa.go:187]: Usually tsh doesn't filter suggestions, instead it lets the backend reject unwanted device types. That said, this is looking wonky, let me try to do better.

Comment (codingllama) [reply tool/tsh/mfa.go]: I attempted to deal with the various nuances of second factor configuration and decided, in the end, that is was not worth it. We now use preferredMFA only to pick between U2F and Webauthn and let the backend deal with the rest.

Let me know if it looks better.

File: lib/client/mfa.go [hunk 2]
```diff
@@ -145,7 +139,7 @@ func PromptMFAChallenge(ctx context.Context, proxyAddr string, c *proto.MFAAuthe
 		}
 	}
 	return nil, trace.BadParameter(
-		"failed to authenticate using all U2F and TOTP devices, rerun the command with '-d' to see error details for each device")
+		"failed to authenticate using all MFA and TOTP devices, rerun the command with '-d' to see error details for each device")
```
Comment (zmb3) [inline lib/client/mfa.go:142]: Isn't TOTP technically an MFA device too?

File: tool/tsh/mfa.go [hunk 3]
```diff
@@ -43,14 +45,17 @@ import (
 	wancli "github.com/gravitational/teleport/lib/auth/webauthncli"
 )
 
+// tshDeviceType represents the device types that appear in the tsh CLI.
+type tshDeviceType = string
```
Comment (zmb3) [inline tool/tsh/mfa.go:49]:
```suggestion
type tshDeviceType string
```

No need for a type alias here.

Comment (codingllama) [reply tool/tsh/mfa.go:49]: Dropping the alias gets rather painful, as we won't be able to use tshDeviceType directly as a string anymore (which is kind of its reason to be).

File: lib/client/mfa.go (hunk 2 above)
Comment (codingllama) [reply lib/client/mfa.go:142]: True, should I drop the "and TOTP" part of it?

**Comment by codingllama** [comment]
Friendly ping?

**Review by kimlisa — APPROVED** [review]

File: tool/tsh/mfa.go [hunk 4]
```diff
@@ -45,14 +45,17 @@ import (
...
+
 const (
-	totpDeviceType     = "TOTP"
-	u2fDeviceType      = "U2F"
-	webauthnDeviceType = "WEBAUTHN"
+	totpDeviceType     tshDeviceType = "TOTP"
+	u2fDeviceType                    = "U2F"
+	webauthnDeviceType               = "WEBAUTHN"
```
[the remark this answers is not in the document]
Comment (codingllama) [reply tool/tsh/mfa.go:54]: Well spotted. My brain somehow thought there was implicit repetition there.

File: lib/client/mfa.go (hunk 2 above)
Comment (zmb3) [reply lib/client/mfa.go:142]: I would.

File: tool/tsh/mfa.go (hunk 3 above)
Comment (zmb3) [reply tool/tsh/mfa.go:49]: Then why have the alias at all? You're not getting anything extra out of the type system, so at that point why not just use `string` directly?

File: lib/client/mfa.go (hunk 2 above)
Comment (codingllama) [reply lib/client/mfa.go:142]: Done, changed to `failed to authenticate using all MFA devices, (...)`.

File: tool/tsh/mfa.go (hunk 3 above)
Comment (codingllama) [reply tool/tsh/mfa.go:49]: Fair enough, dropped.

**Review by zmb3 — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**